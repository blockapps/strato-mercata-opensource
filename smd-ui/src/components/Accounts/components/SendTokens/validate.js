import { isOauthEnabled } from "../../../../lib/checkMode";

export const validate = (values) => {
  const errors = {};

  if (!values.from) {
    errors.value = 'Please select a user';
  }

  if (!values.fromAddress) {
    errors.value = 'Please select a address';
  }

  if (isOauthEnabled() && !values.address) {
    errors.value = "Please enter address"
  }

  if (!isOauthEnabled()) {

    if (!values.password) {
      errors.value = 'Please enter a password';
    }
    if (!values.radio && !values.toAddress) {
      errors.value = "Please select address"
    }

    if (values.radio === "0" && !values.toAddress) {
      errors.value = "Please select address"
    }

    if (values.radio === "1" && !values.address) {
      errors.value = "Please enter address"
    }
  }

  if (!values.value) {
    errors.value = 'Please enter a value';
  }

  return errors;
};

export default validate;
