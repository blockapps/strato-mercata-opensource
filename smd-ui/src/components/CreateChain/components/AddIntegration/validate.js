export function validate(values) {
  const errors = {};

  return errors;
}