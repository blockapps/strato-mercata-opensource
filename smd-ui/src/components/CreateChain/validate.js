export function validate(values) {
  const errors = {};

  if (!values.chainName) {
    errors.chainName = 'required';
  }

  if (values.members && !values.members.length) {
    errors.members = 'at least add one member';
  }

  return errors;
}