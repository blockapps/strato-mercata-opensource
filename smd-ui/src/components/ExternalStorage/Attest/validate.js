import { isOauthEnabled } from "../../../lib/checkMode";

export default function validate(values) {
  const errors = {};

  if (!isOauthEnabled() && !values.username) {
    errors.username = 'Select username';
  }

  if (!isOauthEnabled() && !values.address) {
    errors.address = 'Address required';
  }

  if (!isOauthEnabled() && !values.password) {
    errors.password = 'Password required';
  }

  if (!values.contractAddress) {
    errors.contractAddress = "Can't be blank";
  }

  return errors;
}