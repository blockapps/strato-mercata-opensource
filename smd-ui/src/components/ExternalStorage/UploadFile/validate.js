import { isOauthEnabled } from "../../../lib/checkMode";

export function validate(values) {

  const errors = {};

  if (!isOauthEnabled() && !values.username) {
    errors.username = 'Select username';
  }

  if (!isOauthEnabled() && !values.address) {
    errors.address = 'Address required';
  }

  if (!isOauthEnabled() && !values.password) {
    errors.password = 'Password required';
  }

  if (!values.content) {
    errors.content = 'Attach file';
  }

  if (!values.provider) {
    errors.provider = 'Select provider';
  }

  if (!values.description) {
    errors.description = 'Attach file';
  }

  return errors;
}