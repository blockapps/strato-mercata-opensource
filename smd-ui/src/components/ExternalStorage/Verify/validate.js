export default function validate(values) {
  const errors = {};

  if (!values.contractAddress) {
    errors.contractAddress = "Can't be blank";
  }

  return errors;
}