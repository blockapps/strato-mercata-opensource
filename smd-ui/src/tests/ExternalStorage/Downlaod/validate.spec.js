import validate from '../../../components/ExternalStorage/Download/validate';

describe('Downlaod: validate', () => {

  test('with values', () => {
    const values = { contractAddress: '9c22ec56dd721cd3ca138dc1d1a05d567e019e36' };
    expect(validate(values)).toMatchSnapshot();
  });

  test('with empty values', () => {
    const values = {};
    expect(validate(values)).toMatchSnapshot();
  });

});