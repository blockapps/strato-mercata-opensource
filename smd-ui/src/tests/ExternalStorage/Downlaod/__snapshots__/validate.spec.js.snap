// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downlaod: validate with empty values 1`] = `
Object {
  "contractAddress": "Can't be blank",
}
`;

exports[`Downlaod: validate with values 1`] = `Object {}`;
